packages/salesforce/src/queryRecordFactory.ts

```diff
@@ -164,6 +170,22 @@ export class RecordFactory {
             ?? name;
     }
 
+    private static transformValue(value: string | number | boolean | undefined): any {
+        if (typeof value === 'string') {      
+            //string matching iso8601Pattern as Date      
+            if (/^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(.\d+)?(([+-]\d{2}\d{2})|Z)?)?$/i.test(value)) {
```

Comment:
Use Luxon Datetime for parsin ISO8601; 
```js
if (/^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(.\d+)?(([+-]\d{2}\d{2})|Z)?)?$/i.test(value)) {
     return DateTime.fromISO(value).toJSDate()
}
```

Consider also checking validity before parsing
```js
if (/^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(.\d+)?(([+-]\d{2}\d{2})|Z)?)?$/i.test(value)) {
     const dateValue = DateTime.fromISO(value);
     if (dateValue.isValid) {
         return dateValue;
     }
}
```